src/lem/eval.rs

```diff
@@ -1,60 +1,99 @@
-use crate::lem;
+use crate::func;
 
-use super::LEM;
-use anyhow::Result;
+use super::Func;
 
-/// Lurk's step function encoded as a LEM
+/// Lurk's step function
 #[allow(dead_code)]
-pub(crate) fn step() -> Result<LEM> {
-    lem!(expr_in env_in cont_in {
+pub(crate) fn eval_step() -> Func {
+    let reduce = reduce();
+    let apply_cont = apply_cont();
+    let make_thunk = make_thunk();
+
+    func!((expr_in, env_in, cont_in): 3 => {
```

Comment:
Can the macro be smarter and define those sizes for us? Of course, not for this PR